src/wazuh_modules/agent_upgrade/wm_agent_parsing.c

```diff
@@ -0,0 +1,143 @@
+/*
+ * Wazuh Module for Agent Upgrading
+ * Copyright (C) 2015-2020, Wazuh Inc.
+ * July 3, 2020.
+ *
+ * This program is free software; you can redistribute it
+ * and/or modify it under the terms of the GNU General Public
+ * License (version 2) as published by the FSF - Free Software
+ * Foundation.
+ */
+
+#include "wazuh_modules/wmodules.h"
+#include "wm_agent_parsing.h"
+#include "wm_agent_upgrade_tasks.h"
+
+int wm_agent_parse_command(const char* buffer, cJSON** json_api, cJSON** params, cJSON** agents) {
+    int retval = -1;
+    cJSON * root = cJSON_Parse(buffer);
+    if (!root) {
+        mterror(WM_AGENT_UPGRADE_LOGTAG, WM_UPGRADE_JSON_PARSE_ERROR,  buffer);
+        *json_api = wm_agent_parse_response_mesage(PARSING_ERROR, upgrade_error_codes[PARSING_ERROR], NULL, NULL, NULL);
+    } else {
+        cJSON *command = cJSON_GetObjectItem(root, "command");
+        *params = cJSON_GetObjectItem(root, "params");
+        *agents = cJSON_GetObjectItem(root, "agents");
+        if (command && *agents) {
+            *json_api = root;
+            if (strcmp(command->valuestring, WM_AGENT_UPGRADE_COMMAND_NAME) == 0) {
+                retval = 0;
```

Comment:
I think it would be more easy to understand if you create an enumerate to assign the command code.